packages/p2p/src/components/app.jsx

```diff
@@ -37,8 +46,18 @@ class App extends Component {
         }
     }
 
+    redirectTo = (path_name, params) => {
+        this.setState({ active_index: path[path_name], parameters: params });
+    }
+
     render() {
-        const { active_index } = this.state;
+        const { active_index, parameters } = this.state;
+        const { currency, is_virtual } = this.props.client;
```

Comment:
we need to add these to the proptypes as well + we could set all the proptypes to app to isRequired